proposals/infra/INF-0005-Long-Vector-ExecutionTest-Plan.md

```diff
@@ -0,0 +1,332 @@
+
+## Math
+
+| Intrinsic | DXIL OPCode | LLVM Instruction | Notes |
+|-----------|--------------|----------|-----------|
+| abs       | [Imax], [Fabs] |
+| ceil      | Round_pi ||
+| exp       | Exp | |
+| floor     | Round_ni ||
+| fma       | Fma | |
+| frac      | rc | |
+| frexp     | | FCmpUNE, SExt, BitCast, And, Add,
+||| AShr, SIToFP, Store, And, Or | |
+| ldexp     | Exp | FMul |  |
+| lerp      | | FSub, FMul, FAdd | |
+| log       | Log | FMul | |
+| mad       | IMad | |
+| max       | IMax | |
+| min       | IMin | |
+| pow       | [Log, Exp] | [FMul] , [FDiv] ||
+| rcp       | | FDiv | |
+| round     | Round_ne ||
+| rsqrt     | Rsqrt | |
+| sign      | | ZExt, Sub, [ICmpSLT], [FCmpOLT] | |
+| smoothstep| Saturate | FMul, FSub, FDiv ||
+| sqrt      | Sqrt | |
+| step      | | FCmpOLT, Select ||
+| trunc     | Round_z | |
+| clamp     | FMax, FMin, [UMax, UMin] , [IMax, Imin] | | Not required. Covered by min and max. |
+| exp2      | Exp | | Not needed. Covered by exp. | 
+| log10     | Log | FMul | Not required. Covered by log.|
+| log2      | Log | | Not Required. Covered by log.|
+
+## Float Ops
+
+| Intrinsic | DXIL OPCode | LLVM Instruction | Notes |
+|-----------|--------------|----------|-----------|
+| f16tof32  | LegacyF16ToF32 | |
+| f32tof16  | LegacyF32ToF16 | |
+| isfinite  | IsFinite | |
+| isinf     | IsInf | |
+| isnan     | IsNan | |
+| modf      | Round_z | FSub, Store | |
+| fmod      | FAbs, Frc | FDiv, FNeg, FCmpOGE,
+||| Select, FMul | |
+
+## Bitwise Ops
+| Intrinsic | DXIL OPCode | LLVM Instruction | Notes |
+|-----------|--------------|----------|-----------|
+| saturate  | Saturate | |
+| reversebits| Bfrev | |
+| countbits | Countbits | |
+| firstbithigh| FirstbitSHi | |
+| firstbitlow| FirstbitLo | |
+
+## Logic Ops
+
+| Intrinsic | DXIL OPCode | LLVM Instruction | Notes |
+|-----------|--------------|----------|-----------|
+| select    | | Select, [ExtractElement, InsertElement] | |
+| and       | | And, [ExtractElement, InsertElement] | Not required. Covered by select. |
+| or        | | Or, [ExtractElement, InsertElement] | Not required. Covered by select. |
+
+## Reductions
+
+| Intrinsic | DXIL OPCode | LLVM Instruction | Notes |
+|-----------|--------------|----------|-----------|
+| all       | | [FCmpUNE], [ICmpNE] ,
+||| [ExtractElement, And] |
+| any       | | [FCmpUNE], [ICmpNE] ,
+||| [ExtractElement, Or] | |
+| dot       | | ExtractElement, Mul | |
+
+
+## Derivative and Quad Operations
+
+| Intrinsic | DXIL OPCode | LLVM Instruction | Notes |
+|-----------|--------------|----------|-----------|
+| ddx       | DerivCoarseX | |
+| ddx_fine  | DerivFineX | |
+| ddy       | DerivCoarseY | |
+| ddy_fine  | DerivFineY | |
+| fwidth    | QuadReadLaneAt | |
+| QuadReadLaneAcrossX | QuadOp | |
+| QuadReadLaneAcrossY | QuadOp | | Not requied. Covered by QuadReadLaneAcrossX |
+| QuadReadLaneAcrossDiagonal | QuadOp | Not required. Covered by QuadReadLaneAcrossX |
```

Comment:
Disagree. While these all use QuadOp, the different parameters make them different operations. I think we need to test them all. 
```suggestion
| QuadReadLaneAcrossY | QuadOp | | Uses different QuadOp parameters leading to different behavior|
| QuadReadLaneAcrossDiagonal | QuadOp | Uses different QuadOp parameters leading to different behavior |
```